packages/block-editor/src/components/block-rename/rename-control.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { useBlockDisplayInformation } from '..';
+import isEmptyString from './is-empty-string';
+import BlockRenameModal from './modal';
+
+export default function BlockRenameControl( { clientId } ) {
+	const [ renamingBlock, setRenamingBlock ] = useState( false );
+
+	const { metadata } = useSelect(
+		( select ) => {
+			const { getBlockAttributes } = select( blockEditorStore );
+
+			const _metadata = getBlockAttributes( clientId )?.metadata;
+			return {
+				metadata: _metadata,
+			};
+		},
+		[ clientId ]
+	);
```

Comment:
Is there a reason to return an object, or is it just to match what we do elsewhere and make it easier to add more properties in the future?

```suggestion
	const metadata = useSelect(
		( select ) => {
			const { getBlockAttributes } = select( blockEditorStore );

			return getBlockAttributes( clientId )?.metadata;
		},
		[ clientId ]
	);
```

Reply:
See https://jardasn.blog/2021/01/22/some-best-practices-for-using-useselect-from-wordpress-data/comment-page-1/#:~:text=%7D-,Prefer,-returning%20objects%20with